common/peer_failed.c

```diff
@@ -56,10 +56,22 @@ void peer_failed_received_errmsg(struct per_peer_state *pps,
 {
 	static const struct channel_id all_channels;
 	u8 *msg;
+	bool sync_error;
+
+	/* <+roasbeef> rusty: sync error can just be a timing thing
+	 * <+roasbeef> andn doesn't always mean that we can't continue forwrd,
+	 *             or y'all sent the wrong info
+	 */
+	/* So while LND is sitting in the corner eating paint, back off. */
```

Comment:
Very unprofessional language there, rusty. What a shame, only to find out it's your bug and the `little snowflake` has to help you out.

Reply:
Is it c-lightning bug? `We close the channel unilaterally because the BOLT specs that is what should be done if we receive an error message`

Reply:
It is both side bug.

1.  C-Lightning should reestablish quickly on reconnection.
2.  LND should not throw out `error` like candy.

1 above is arguably "fine" since we cannot control the resources available on every node C-Lightning can run on anyway, so *some* number of open channels and *some* level of hardware will experience this.

2 is harder to justify, given BOLT correctness.

Reply:
You all keep saying LND should make it a "soft" error or just "hang up" but rusty said "there are no "soft" errors in the protocol":  https://github.com/ElementsProject/lightning/issues/2847#issuecomment-516708753 
So instead of looking at your own code and questioning yourself first that maybe it's your issue you blamed on LND and even called name... 
Don't you ever ask why this mass channel closing doesn't happen so often with other impls but only with c-lightning? And this issue still happens after almost two yrs.  I was talking pavlenex, aka 'bitcoinshirt', on lnd slack a few weeks ago that i opened a channel to his node and it got closed within a few minutes. He told me it hadn't happened anymore, and bam it happened again, lmao ...  
I had this convo with roasbeef on the slack and here's what he said, I hope you'll read it and think about it: 

![image](https://user-images.githubusercontent.com/21127776/62272187-7fe59080-b408-11e9-91ad-e295c3a4fdf6.png)

Reply:
Maybe LND should update the BOLT before deviating from it.

Reply:
Again: the specs says that on receipt of `error` we should force-close.  C-Lightning follows the BOLT standard.  If LND wants to change the behavior of `error` it should negotiate a change in specifications first.  The only reason this appears "normal" to you is because LND has a much bigger advertising budget and thereby runs 85% of the public network.

Just because some application behaves some way and is popular does not make it "correct".  Follow the agreed-upon specifications or at least negotiate for a proper specifications change.  LND is clearly not spec-compliant here, whereas failing a non-spec-defined timeout does not make C-Lightning incorrect.

https://github.com/lightningnetwork/lightning-rfc/blob/master/02-peer-protocol.md#message-retransmission

The above does not mention any timeouts regarding `channel_reestablish`.

1.  LND is not spec-compliant 1: it imposes a time limit on `channel_reestablish` where none is specified.
2.  LND is not spec-compliant 2: it throws a an `error` with `"sync error"` in this case.
3.  LND is not spec compliant 3: if it receives an `error` with `"sync error"` it does not close the channel: https://github.com/lightningnetwork/lightning-rfc/blob/master/01-messaging.md#the-error-message

> The receiving node:
>   - upon receiving `error`:
>     - MUST fail the channel referred to by the error message, if that channel is with the sending node.

LND is clearly in the wrong, despite its popularity.

In any case, I will start to ignore what you say about this topic.

Reply:
The godwin point of compatibility: LND is the Internet Explorer of BOLT

Reply:
>LND is the Internet Explorer of BOLT

That is arguably worse than calling it "precious little snowflake".